components/schedule-editor/src/ScheduleEditor.svelte

```diff
@@ -248,10 +288,165 @@
+
+  type CustomFieldWithId = CustomField & { id: string };
+  let draggedField: CustomFieldWithId | null = null;
+  let draggedFieldIndex: number;
+  let tablerowRect: DOMRect;
+  let tablecellRect: DOMRect[] = []; // To replicate cell dimensions on drag-placeholder
+  type CustomFieldRefs = Record<string, HTMLElement>;
+  let customFieldRefs: CustomFieldRefs = {}; // store the refs to dom nodes once so we can recalculate sizes on the fly
+  let customFieldDomRects: DOMRect[] = [];
+  let customFieldIdOrder: string[] = [];
+  let maxDragTop: number;
+  let maxDragBottom: number;
+  let dragBegins: boolean = false;
+  let dragPlaceholder = {
+    left: 0,
+    top: 0,
+  };
+
+  type CustomFieldDrag = {
+    event: MouseEvent;
+    field: CustomFieldWithId;
+    i: number;
+  };
+
+  function handleCustomFieldDrag({ event, field, i }: CustomFieldDrag) {
+    if (!dragBegins) {
+      customFieldDomRects = getDomRects(customFieldRefs);
+      dragBegins = true;
+    }
+
+    draggedFieldIndex = i;
+    draggedField = field;
+
+    setDragRowPosition(event);
+  }
+
+  function handleCustomFieldDragRelease() {
+    if (draggedField) {
+      draggedField = null;
+    }
+  }
+
+  function swapHoveredWithDraggedFields(e: MouseEvent) {
+    const hoveredFieldIndex = customFieldDomRects.findIndex(
+      (domRect) => domRect.top < e.clientY && e.clientY < domRect.bottom,
+    );
+
+    const hoveredFieldId = customFieldIdOrder[hoveredFieldIndex];
+
+    const fieldSwapRequired =
+      hoveredFieldIndex >= 0 && hoveredFieldId !== draggedField?.id;
+
+    if (fieldSwapRequired) {
+      let updatedFieldOrder = [..._this.custom_fields] as CustomFieldWithId[];
+
+      // Swap the current hovered field with the dragged field
+      let temp = updatedFieldOrder[hoveredFieldIndex];
+      updatedFieldOrder[hoveredFieldIndex] =
+        updatedFieldOrder[draggedFieldIndex];
+      updatedFieldOrder[draggedFieldIndex] = temp;
+
+      // update custom fields to reflect the drag order
+      _this.custom_fields = updatedFieldOrder;
+      customFieldIdOrder = updatedFieldOrder.map((field) => field.id);
+
+      // current dragged index is now what was hovered
+      draggedFieldIndex = updatedFieldOrder.findIndex(
+        (customField) => customField.id === draggedField?.id,
+      );
+    }
+  }
+
+  const throttledDraggedFieldSwap = throttle(swapHoveredWithDraggedFields, 100);
+
+  const setDragRowPosition = (e: MouseEvent) => {
+    const rowHeightOffset = tablerowRect.height / 2;
+    const dragRowPosition = e.pageY;
+
+    // stick the drag preview if cursor is beyond max top and max bottom
+    let previewTopPosition: number | null = null;
+    if (dragRowPosition <= maxDragTop) {
+      previewTopPosition = maxDragTop - rowHeightOffset;
+    } else if (dragRowPosition >= maxDragBottom) {
+      previewTopPosition = maxDragBottom - rowHeightOffset;
+    }
+
+    // set initial position of drag preview
+    dragPlaceholder = {
+      ...dragPlaceholder,
```

Comment:
nit - Don't think we need to spread `dragPlaceholder` here since we're setting `left` and `top` already